File: IT/Strutture dati/Esercizi/C++/esercizio20.cpp
/* file esercizio20.cpp */

#include <iostream>
#include <iomanip>

using std::cout;
using std::endl;
using std::left;
using std::setw;

int main()
{
	const int M = 3;
	const int N = 3;
	int matrix[M][N];

	int k = M * N;
	for (int i = 0; i < M; i++) {
		for (int j = 0; j < N; j++) {
			matrix[i][j] = k;
			k--;
		}
	}

	for (int i = 0; i < M; i++) {
		for (int j = 0; j < N; j++) {
			cout << left << setw(2) << matrix[i][j];
		}
		cout << endl;
	}
}

File: IT/1/Strutture dati/Esercizi/Java/Esercizio20.java
/* file Esercizio20.java */

public class Esercizio20 {
	public static void main(String [] args) {
		final int M = 3;
		final int N = 3;
		int [][] matrix = new int[M][N];

		int k = M * N;
		for (int i = 0; i < matrix.length; i++) {
			for (int j = 0; j < matrix[i].length; j++) {
				matrix[i][j] = k;
				k--;
			}
		}

		for (int i = 0; i < matrix.length; i++) {
			for (int j = 0; j < matrix[i].length; j++) {
				System.out.printf("%-2d", matrix[i][j]);
			}
			System.out.print(System.lineSeparator());
		}
	}
}

